src/libcore/option.rs

```diff
@@ -1040,6 +1040,25 @@ fn expect_failed(msg: &str) -> ! {
 // Trait implementations
 /////////////////////////////////////////////////////////////////////////////
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<T: Clone> Clone for Option<T> {
+    #[inline]
+    fn clone(&self) -> Self {
+        match self {
+            &Some(ref x) => Some(x.clone()),
```

Comment:
Prefer using default match bindings (and in the other functions as well).

Reply:
Done